### PR TITLE
Migrate the performance page to use the Perfetto VM timeline API

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This allows us to include CPU samples in the timeline. Fixes https://github.com/flutter/devtools/issues/4755.
Screenshot 2024-03-07 at 10 05 14 AM

This PR removes our code that was needed for processing the Chrome Trace Format and replaces it with code to deal with the protozero perfetto format.

After this lands, I will follow on with fixes for these two issues to make sure this change lands well:
- https://github.com/flutter/devtools/issues/7334
- https://github.com/flutter/devtools/issues/7315

FYI @a-siva @bkonyi @derekxu16

### CODE REVIEW

**Comment by bkonyi** [comment]
Overall, things are looking good!

File: packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart [hunk 1]
```diff
@@ -801,7 +801,7 @@ class CpuProfileMetaData extends ProfileMetaData {
   }
 }
 
-class CpuSampleEvent extends TraceEvent {
+class CpuSampleEvent extends ChromeTraceEvent {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart:804]: @bkonyi @derekxu16 general question - will the VM Service CPU profiler APIs also be moving away from the Chrome trace event format at some point?

File: packages/devtools_app/test/performance/timeline_events/perfetto/tracing_model_test.dart [hunk 2]
```diff
@@ -0,0 +1,134 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
```
Comment (bkonyi) [inline packages/devtools_app/test/performance/timeline_events/perfetto/tracing_model_test.dart:1]: Does this copyright date need to be updated?

File: packages/devtools_app/test/test_infra/utils/test_utils.dart [hunk 3]
```diff
@@ -7,20 +7,45 @@ import 'dart:convert';
...
 
-int _testTimeReceived = 0;
-TraceEventWrapper testTraceEventWrapper(Map<String, dynamic> json) {
-  return TraceEventWrapper(testTraceEvent(json), _testTimeReceived++);
+  return FlutterTimelineEvent(mockFirstTrackEvent)
+    ..addEndTrackEvent(mockEndTrackEvent);
 }
 
+ChromeTraceEvent testTraceEvent(Map<String, dynamic> json) =>
+    ChromeTraceEvent(jsonDecode(jsonEncode(json)));
```
Comment (bkonyi) [inline packages/devtools_app/test/test_infra/utils/test_utils.dart:47]: What's the deal with the encode-decode here?

File: packages/devtools_app/test/test_infra/utils/test_utils.dart [hunk 4]
```diff
@@ -7,20 +7,45 @@ import 'dart:convert';
...
 
 import 'package:devtools_app/devtools_app.dart';
+import 'package:devtools_test/devtools_test.dart';
 import 'package:flutter/services.dart';
 import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
 
-SyncTimelineEvent testSyncTimelineEvent(TraceEventWrapper eventWrapper) =>
-    SyncTimelineEvent(eventWrapper);
+FlutterTimelineEvent testTimelineEvent({
```
Comment (bkonyi) [inline packages/devtools_app/test/test_infra/utils/test_utils.dart:15]: Does this need some documentation?

**Review by bkonyi — APPROVED** [review]

File: packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart (hunk 1 above)
Comment (bkonyi) [reply packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart:804]: Maybe, but there's no plans to drop support currently on our timeline.

**Comment by kenzieschmoll** [comment]
I will land this and follow up with a few improvements.

**Pull request merged**